### ISSUE TITLE
KeePassXC 2.5.3 crashes on launch on macOS 10.15.4

### ISSUE THREAD

**Issue by grahampugh** [keepassxreboot/keepassxc#4398]
## Expected Behavior
KeePassXC application opens as normal

## Current Behavior
Application crashes immediately upon launch. A system dialog is opened (see attached image).
![Screenshot 2020-03-03 at 21 55 40](https://user-images.githubusercontent.com/5802725/75819407-e9f95600-5d9a-11ea-930d-e3019fef527e.png)

## Steps to Reproduce
1. Upgrade to, or freshly install macOS 10.15.4 beta 3 on a Mac computer
2. Install KeePassXC by downloading from the keepassxc.org website.
3. Attempt to open the application.

## Context
Note that KeePassX runs without problems on this same beta OS.

## Debug Info
Cannot post output from app as it will not open
KeePassXC - 2.5.3
Revision: current release version

Operating system: macOS 10.15.4 beta 3
CPU architecture: x64
Kernel: Darwin

Enabled extensions:
- None (fresh installation)

**Comment by grahampugh** [comment]
Additional info: the traceback is as follows:
```
Process:               KeePassXC [5529]
Path:                  /Applications/KeePassXC.app/Contents/MacOS/KeePassXC
Identifier:            KeePassXC
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        bash [5523]
Responsible:           Terminal [2020]
User ID:               501

Date/Time:             2020-03-03 21:58:47.824 +0100
OS Version:            Mac OS X 10.15.4 (19E242d)
Report Version:        12
Anonymous UUID:        6E05C5B7-3DDF-F9BE-1403-DAAE1D80F66B

Sleep/Wake UUID:       3FBD8310-DB6C-4000-B866-CCB1B8D6A053

Time Awake Since Boot: 23000 seconds
Time Since Wake:       1200 seconds

System Integrity Protection: enabled

Crashed Thread:        Unknown

Exception Type:        EXC_CRASH (Code Signature Invalid)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace CODESIGNING, Code 0x1

kernel messages:

Backtrace not available

Unknown thread crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000000  rbx: 0x0000000000000000  rcx: 0x0000000000000000  rdx: 0x0000000000000000
  rdi: 0x0000000000000000  rsi: 0x0000000000000000  rbp: 0x0000000000000000  rsp: 0x00007ffedfc28af8
   r8: 0x0000000000000000   r9: 0x0000000000000000  r10: 0x0000000000000000  r11: 0x0000000000000000
  r12: 0x0000000000000000  r13: 0x0000000000000000  r14: 0x0000000000000000  r15: 0x0000000000000000
  rip: 0x000000011ac8a000  rfl: 0x0000000000000200  cr2: 0x0000000000000000
  
Logical CPU:     0
Error Code:      0x00000000
Trap Number:     0


Binary images description not available


External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 26249
    thread_create: 0
    thread_set_state: 0

Model: MacBookAir6,1, BootROM 119.0.0.0.0, 2 processors, Dual-Core Intel Core i7, 1.7 GHz, 8 GB, SMC 2.12f143
Graphics: kHW_IntelHD5000Item, Intel HD Graphics 5000, spdisplays_builtin
Memory Module: BANK 0/DIMM0, 4 GB, DDR3, 1600 MHz, 0x80AD, 0x483943434E4E4E384A544D4C41522D4E544D
Memory Module: BANK 1/DIMM0, 4 GB, DDR3, 1600 MHz, 0x80AD, 0x483943434E4E4E384A544D4C41522D4E544D
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0x117), Broadcom BCM43xx 1.0 (7.77.111.1 AirPortDriverBrcmNIC-1600.4)
Bluetooth: Version 7.0.4f3, 3 services, 27 devices, 1 incoming serial ports
Network Service: Wi-Fi, AirPort, en0
Serial ATA Device: APPLE SSD TS0128F, 121.33 GB
USB Device: USB 3.0 Bus
USB Device: BRCM20702 Hub
USB Device: Bluetooth USB Host Controller
Thunderbolt Bus: MacBook Air, Apple Inc., 23.6
```

**Comment by droidmonkey** [comment]
This is an issue with the beta update. It does not verify properly signed apps.

**Comment by grahampugh** [comment]
Interesting, I haven’t seen that reported anywhere else such as the Apple Developer Forums or the MacAdmins Slack. Nor is it listed as a known issue in the Developer Notes. Let’s hope it’s fixed before release. 

As an update, it is not fixed in Beta 4, which was released since I raised the issue.

**Comment by phoerious** [comment]
Please try a snapshot build from https://snapshot.keepassxc.org.

**Comment by grahampugh** [comment]
@phoerious The latest snapshot can be opened, although it is not signed.

**Comment by phoerious** [comment]
Then it's probably a Qt bug or something.

**Comment by droidmonkey** [comment]
I doubt it's Qt. This is related to signed application not being verifiable. See this line in the trace: `Exception Type:        EXC_CRASH (Code Signature Invalid)`

**Comment by phoerious** [comment]
Maybe the certificate isn't accepted or so.

**Comment by spluess** [comment]
FYI, same happens here on normal Mojave 10.14.6
<img width="420" alt="keypassxc-253-crash-mojave" src="https://user-images.githubusercontent.com/13042794/75918321-d06c1300-5e5b-11ea-80cd-25787084efa2.png">

```
Process:               KeePassXC [895]
Path:                  /Applications/KeePassXC.app/Contents/MacOS/KeePassXC
Identifier:            org.keepassx.keepassxc
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           KeePassXC [895]
User ID:               501

Date/Time:             2020-03-04 21:03:24.429 +0100
OS Version:            Mac OS X 10.14.6 (18G4017)
Report Version:        12
Anonymous UUID:        4336434E-009E-9B9B-3EE1-FAFCCCF6C984

Sleep/Wake UUID:       2FBBDEAC-0362-4F2F-8A8F-4351BB9A16D8

Time Awake Since Boot: 1400 seconds
Time Since Wake:       490 seconds

System Integrity Protection: enabled

Crashed Thread:        0

Exception Type:        EXC_CRASH (Code Signature Invalid)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace CODESIGNING, Code 0x1

kernel messages:

VM Regions Near 0 (cr2):
--> 
    __TEXT                 0000000108f23000-00000001092b5000 [ 3656K] r-x/r-x SM=COW  

Thread 0 Crashed:
0                                 	0x000000010d980000 _dyld_start + 0

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000000  rbx: 0x0000000000000000  rcx: 0x0000000000000000  rdx: 0x0000000000000000
  rdi: 0x0000000000000000  rsi: 0x0000000000000000  rbp: 0x0000000000000000  rsp: 0x00007ffee6cdcbc0
   r8: 0x0000000000000000   r9: 0x0000000000000000  r10: 0x0000000000000000  r11: 0x0000000000000000
  r12: 0x0000000000000000  r13: 0x0000000000000000  r14: 0x0000000000000000  r15: 0x0000000000000000
  rip: 0x000000010d980000  rfl: 0x0000000000000200  cr2: 0x0000000000000000
  
Logical CPU:     0
Error Code:      0x00000000
Trap Number:     0


Binary Images:
       0x108f23000 -        0x1092b4fff + (0) <16AE8CED-0E4C-3291-AE43-913B0D7192FF> 
       0x10d97f000 -        0x10d9e970f + (655.1.1) <C192CA31-D059-3770-9882-D864FEFA0C96> 

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 848
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=4552K resident=0K(0%) swapped_out_or_unallocated=4552K(100%)
Writable regions: Total=8404K written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=8404K(100%)
 
                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
STACK GUARD                       56.0M        1 
Stack                             8192K        1 
__DATA                             372K        4 
__LINKEDIT                         468K        2 
__TEXT                            4084K        2 
shared memory                        8K        2 
===========                     =======  ======= 
TOTAL                             68.8M       12 

Model: MacBookPro11,3, BootROM 158.0.0.0.0, 4 processors, Intel Core i7, 2.6 GHz, 16 GB, SMC 2.19f12
Graphics: kHW_IntelIrisProItem, Intel Iris Pro, spdisplays_builtin
Graphics: kHW_NVidiaGeForceGTX750MItem, NVIDIA GeForce GT 750M, spdisplays_pcie_device, 2 GB
Memory Module: BANK 0/DIMM0, 8 GB, DDR3, 1600 MHz, 0x80AD, 0x484D54343147533641465238412D50422020
Memory Module: BANK 1/DIMM0, 8 GB, DDR3, 1600 MHz, 0x80AD, 0x484D54343147533641465238412D50422020
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0x134), Broadcom BCM43xx 1.0 (7.77.61.3 AirPortDriverBrcmNIC-1305.9)
Bluetooth: Version 6.0.14d6, 3 services, 27 devices, 1 incoming serial ports
Network Service: Wi-Fi, AirPort, en0
Serial ATA Device: APPLE SSD SM0512F, 500.28 GB
USB Device: USB 3.0 Bus
USB Device: Apple Internal Keyboard / Trackpad
USB Device: BRCM20702 Hub
USB Device: Bluetooth USB Host Controller
Thunderbolt Bus: MacBook Pro, Apple Inc., 17.1
```

**Comment by phoerious** [comment]
I just downloaded our release and tested it on Catalina, works for me. Certificate is valid until 2022.

**Comment by tomhundt** [comment]
I just updated my other MacBook Pro to **MacOS 10.15.4 beta 4** (note, not beta 3), and noted more carefully what happened, in case it helps someone:
- Computer rebooted as part of the update
- Ran KeePassXC, and it crashed. Got the crash report popup.
- Tried again just in case, same thing happened.
- Opened the snapshot .dmg file from the nightly build release (e26063a), which I'd saved from yesterday, when I had this problem on my other machine.
- It popped up the usual Finder window with the app icon and a link to Applications.
- Dragged the KP icon to the Applications folder.
- Got a popup complaining, "macOS cannot verify that this app is free from malware", with the options "Move to Trash" and "Cancel".
- Clicked Cancel.  It kept popping up again, every time I clicked Cancel.
- Opened System Prefs > Security & Privacy > General
There was a message and button waiting there, "Open Anyway".  Clicked it.
- Clicked Cancel on the other one.
- That popup came up again, but now with a different msg and options
    "macOS cannot verify the developer of “KeePassXC.app”. Are you
    sure you want to open it?"
- clicked newly-available Open
- It still doesn't run, but no more popups, either.
- Also on System Prefs there's no prompt on the Security & Privacy page, which I still had open.
- Clicked on Dock KP icon to run it.
- Success, it opens.

**Comment by matthiasschroder** [comment]
Still crashes on beta5. Is the problem understood by now?

**Comment by phoerious** [comment]
I could not reproduce it and the certificate the binary was signed with is valid until 2022.

**Comment by matthiasschroder** [comment]
Hm, so you can run it under macOS 10.15.4 beta{3,4,5} without problems? I tried on three different Macs and it crashes consistently on all three when I try to start it.

**Comment by phoerious** [comment]
"macOS cannot verify that this app is free from malware" sounds like there's something wrong with the stapled notarisation ticket.

**Comment by 317wurst** [comment]
Crash on launch reproduced here. KPxc 2.5.3 public release on Catalina Beta 3 and 4. 
Log looks like grahampugh's.

**Comment by matthiasschroder** [comment]
And the same happens on 10.13.6 with SecUpd 2020-002 beta5.

**Comment by matthiasschroder** [comment]
Just for completeness: same result on 10.14.6 with SecUpd 2020-002 beta5.

**Comment by 317wurst** [comment]
I tried an install from scratch on a fresh 10.15.4 Beta (19E258a) system and got this...
<img width="415" alt="Screenshot 2020-03-16 at 14 36 21" src="https://user-images.githubusercontent.com/62055540/76771442-a2fa6000-679f-11ea-96c0-e4987138ea4d.png">

**Comment by phoerious** [comment]
Snapshot build?

**Comment by 317wurst** [comment]
No. Please read carefully.

> 2.5.3 release

**Comment by grahampugh** [comment]
For info: It’s likely that macOS 10.15.4 will be released this week or next week at the latest, when looking at the pattern of previous years (unless current events delay this). That means KeePassXC will not function on a productive version of macOS unless a release is organised soon. I have personally moved over to KeePassX since I have to work on a beta system. 

As others have intimated, it’s somewhat likely that this is a Notarisation issue.

**Comment by somehotdogman** [comment]
Moved to KeePassX for now also.

**Comment by droidmonkey** [comment]
It is beyond ironic that macOS prefers you to run unsigned software than software that is signed AND notorized by them!

**Comment by matthiasschroder** [comment]
beta 6 is out, still the same issue.

**Comment by droidmonkey** [comment]
@phoerious can you resign and notarize 2.5.3 and post for testing?

**Comment by dpbaker57** [comment]
I downloaded and was able to run the latest 2.6.0 snapshot. That's good but now I can't stop worrying that this pre-release version will trash my vault data. Any chance of a 2.5.4 using a correct code validation?

**Comment by droidmonkey** [comment]
You don't have to worry about corruption, we have not changed how the database is written from 2.5.4 to 2.6.0.

**Comment by grahampugh** [comment]
MacOS 10.15.4 officially dropped today, so KeePassXC now officially does not work on the latest macOS. 

Cheers,
Graham

**Comment by Toh-Bhydill** [comment]
KeyPassXC 2.5.1 functions on MacOS 10.15.4

**Comment by trendchiller** [comment]
same problem here, 2.5.3 crashes on latest MacOS 10.15.4

**Comment by mahnunchik** [comment]
Same problem

**Comment by mikemadden42** [comment]
I'm seeing the same issue on macOS 10.15.4.

<img width="532" alt="Screen Shot 2020-03-24 at 17 20 34" src="https://user-images.githubusercontent.com/5888518/77482415-d8342d00-6df3-11ea-8c8e-b9444f166580.png">
<img width="532" alt="Screen Shot 2020-03-24 at 17 20 22" src="https://user-images.githubusercontent.com/5888518/77482420-de2a0e00-6df3-11ea-8405-5fe125ebb76b.png">
<img width="1112" alt="Screen Shot 2020-03-24 at 17 20 44" src="https://user-images.githubusercontent.com/5888518/77482434-e84c0c80-6df3-11ea-847d-31613ab13da9.png">

**Comment by phoerious** [comment]
We are aware of the problem and are trying to find a solution. Please stop posting "me too". Thanks!

**Comment by phoerious** [comment]
The crash comes due to Apple deciding to change their code signing policies in a f***ing minor release. For some reason the entitlements set on our application cause a crash when we disable the sandbox, which we need to do in order for the browser extension to work (and some other stuff I believe). Until now, a workaround was to specify the sandbox=false entitlement twice, but now Apple decided that they don't accept duplicate entitlement keys anymore, hence this crash now. I can sign the app, but I cannot notarise it anymore (the previous binary is properly notarised, but doesn't work anymore, regardless).

At the moment, I do not see a way to have both a working app and an app that I can notarise. This is so incredibly silly and frustrating. Apple keeps doing stuff like this again and again. I cannot understand how this has ever become a serious platform.

**Comment by swa66** [comment]
Can you please make a point release that's not signed ?

10.15.4 is out and being auto-installed as we speak.

Expecting random users to install a snapshot that also works way different (e.g. I had to manually search my entire database to find my github password as CMD-F doesn't work in the snapshot) is simply not a realistic solution.

**Comment by mikemadden42** [comment]
I had previously used MacPass (https://macpassapp.org/).  It still works on macOS 10.15.4.  It's touted as a _KeePass-compatible password manager for macOS_ so it likely works with existing files.  It works on several I've tried to use.

**Comment by fantom-x** [comment]
> At the moment, I do not see a way to have both a fully functional app and an app that I can notarise and that doesn't crash. 

A commenter above mentioned that `KeyPassXC 2.5.1 and 2.5.2 function on MacOS 10.15.4`: are they using different entitlements?

**Comment by droidmonkey** [comment]
I believe 2.5.3 was the first release we notorized.

**Comment by phoerious** [comment]
I uploaded a new build with the suffix "-2" which fixes this problem. It may, however, come at the cost of TouchID, which I cannot check due to lack of suitable hardware. Please test if TouchID still works and if it doesn't, then there is most likely very little I can do about it.

**Comment by volans-** [comment]
@phoerious thanks a lot for the quick fix. I can confirm the latest `2.5.3-2` just prompts for the usual software downloaded from the internet confirmation and then opens normally.
I don't have either a TouchID device to test that functionality. Sorry.

**Comment by tromlet** [comment]
Oh! So it looks like I'm not the only one with this issue. This reminds me of trying to do... anything... with PowerShell. Anyways! Does your "-2" suffix release get pulled with a `brew cask upgrade`? Because mine's still hard dying. :(

**Comment by liangzan** [comment]
https://github.com/keepassxreboot/keepassxc/releases/download/2.5.3/KeePassXC-2.5.3-2.dmg
Posting the link

**Comment by phoerious** [comment]
Not unless somebody updates the formula.

**Comment by tromlet** [comment]
Fair! Sorry dude. For what it's worth... great app! You've made lots of people's lives much better with it. And then, _when those people can't use it_... D:

**Comment by dicksondickson** [comment]
KeePassXC-2.5.3-2 confirm is working again but TouchID doesn't work (don't care). Thank you for the excellent work! Just donated. Stay safe.

Also, is 2.6.0 snapshot safe to use? Looks great, would love to use it daily.

**Comment by phoerious** [comment]
I use it, but I wouldn't recommend it for primetime to everybody. Definitely keep a backup (keep one anyway).

**Comment by grahampugh** [comment]
Thanks for making the update. The app does seem to work and although TouchID doesn't operate, it would be great to get this pushed to the official release - TouchID is no biggie for me (neither is browser integration since I haven't got that working yet anyway!).

**Comment by andycu2015** [comment]
@phoerious Will you upload 2.5.3-2 to the main website (https://keepassxc.org) so people can just go there and download it?

**Comment by wiedsche** [comment]
I can confirm. 2.5.3-2 starts again and is usable.

But there are some bugs:
- un/locking doesn't work properly.
-- after auto locking on minimize and reopen: I see all my passwords but can not click anything. The title bar says "Locked"
- TouchId doesn't work. No prompt for it.
I think both are related to each other.
- the settings window doesn't open sometimes when calling it in the top menu (not from the icon bar - there it works)

**Comment by MaurizioLoreti** [comment]
From time to time I open https://keepassxc.org/download#mac , and seeing mentioned 2.5.3 as last release, I did not care about clicking the download link - If I had done that, 2.5.3-2 had shown up. If you release a brand new 2.5.3, you should make that CLEAR in your download page. Thank you for your time.

**Comment by phoerious** [comment]
The link on the website is already updated and points to the -2 binary.

**Comment by grahampugh** [comment]
It is unfortunate that the version number of this new version is the same as the old version. This prevents management systems from picking up that there is a change, and so the updated app won't be pushed to clients to fix this problem. Any chance of a version bump to 2.5.4 or 2.5.3.1?

**Comment by phoerious** [comment]
I don't want to rebuild all other platforms and our version check doesn't work for macOS users anyway when they can't start KeePassXC.

**Comment by jograb** [comment]
Or you can switch to MacPass (https://github.com/MacPass/MacPass). Nice side effect: looks better. :)

**Comment by bottee** [comment]
Just ran into this issue after update to MacOS 10.15.4. Luckily there are alternatives to keepassxc. Otherwise I would not have been able to sign-in on any page or program before I found this issue! Also Mac Update doesn't show an update up to now. 
Such a bug needs a visible and detectable update!
Beside this, you are doing a great job!

**Comment by Xefir** [comment]
I tried to install the new version but this message appear and I can't launch the app.

<img width="532" alt="Capture d’écran 2020-03-25 à 11 04 46" src="https://user-images.githubusercontent.com/961976/77524991-97d2be80-6e88-11ea-9161-e63ce7090095.png">

I check if the app is badly download but the DIGEST for the new version and the SHA256 of my file is the same : 

xefir@Tordy:~/Nextcloud/Logitech/Tools$ shasum -a 256 KeePassXC-2.5.3-2.dmg 
e278f02466e3fb294c5c676d7b0ba3c60302d0b08273b8088ff39ca2e97e79af  KeePassXC-2.5.3-2.dmg

Have you an idea ? :(

**Comment by phoerious** [comment]
@bottee We do not maintain MacUpdate. It's up to them to update their downloads.

@Xefir Make sure you download the version ending in -2.

**Comment by schnaider** [comment]
PGP signature and SHA-256 digest belong to a different version. So, I can't check the `...- 2.dmg` version.

**Comment by Xefir** [comment]
It's the latest version with the -2.

I download this app : https://github.com/keepassxreboot/keepassxc/releases/download/2.5.3/KeePassXC-2.5.3-2.dmg
And this DIGEST : https://github.com/keepassxreboot/keepassxc/releases/download/2.5.3/KeePassXC-2.5.3-2.dmg.DIGEST

**Comment by phoerious** [comment]
I separated the Sierra builds from the Mojave+ builds on the downloads page, should be clearer now.

**Comment by Xefir** [comment]
Okay, for those that has the same error message has me, the solution is simple : reboot.
No joke here, it works now.
Thank you for the time and the work you done :)

**Comment by lolzim** [comment]
Hi,
homebrew version KeePassXC-2.5.3 constantly crashes on start after today's security update (Mojave). I uninstalled homebrew version and installed KeePassXC-2.5.3-2 from website. It runs normally. I suppose -2 version should now be pushed to homebrew.

And thanks to the whole team, this app is in my top 5 ;)

**Comment by chriswthomson** [comment]
KeePassXC-2.5.3-2 works for me on macOS Catalina 10.15.4. Thanks for sorting this!

**Comment by tromlet** [comment]
> Or you can switch to MacPass (https://github.com/MacPass/MacPass). Nice side effect: looks better. :)

I switched FROM MacPass... I liked KeePassXC's cross-platform-ness, and the built-in browser integration.

**Comment by thalunil** [comment]
> KeePassXC-2.5.3-2 works for me on macOS Catalina 10.15.4. Thanks for sorting this!

same here - ACK

**Comment by daisukixci** [comment]
Damn sorry to have been part of the duplicated issues 🙇 
Just to know, on which timeline do you plan to update the brew cask (if you planned it :))? to know if I can wait or should us the pkg 🙇

**Comment by maxim-belkin** [comment]
> Just to know, on which timeline do you plan to update the brew cask (if you planned it :))? to know if I can wait or should us the pkg 🙇

https://github.com/Homebrew/homebrew-cask/pull/79282

**Comment by phoerious** [comment]
We do not maintain homebrew.

**Comment by ccampo133** [comment]
Thanks for the quick action on this issue - I just ran into it after upgrading to 10.15.4.

Unfortunately, I still cannot get 2.5.3-2 to work. I downloaded the DMG installer from the link on the main KeePassXC website (verified there is the -2). After installing (dragging to "Applications"), I get a constant notification "KeePassXC" is damaged and can't be opened. You should move it to the the Trash"

<img width="438" alt="Screen Shot 2020-03-25 at 2 45 16 PM" src="https://user-images.githubusercontent.com/957704/77573729-7af9b380-6ea7-11ea-8a61-0d3b095581b9.png">

Clicking cancel just causes this message to appear in a loop. I tried some of the other recommendations here (reboot, change security settings, etc.), and no dice. 

Unfortunately it looks like I will have to revert to an older version until this gets sorted out. Hopefully this is helpful to add another user experience.

**Comment by nmajin** [comment]
Any update? seems like macOS doesn't like the "libecomlodr.dylib" but even removing this file I still cannot start KeePass

**Comment by sjlehn** [comment]
You can make it work (if you don't care about the code signature, do this at your own risk) by simply signing the binary yourself:
`sudo codesign -f -s - /Applications/KeePassXC.app/Contents/MacOS/KeePassXC`

That will get it up and running (mostly - the app itself will launch and operate normally, but the browser integration doesn't...I theorize that the keepassxc-proxy binary might also be necessary for that to work, but I didn't bother experimenting, I'm used to copy-pasting password anyway and needed to get back to other things)

**Comment by droidmonkey** [comment]
Download 2.5.3-2 from our releases page.

**Comment by nmajin** [comment]
> Download 2.5.3-2 from our releases page.

Is it available in homebrew yet? I guess I can overlay this version on top but I don’t know what kind of mess that will result with homebrew.

**Comment by maxim-belkin** [comment]
> > Download 2.5.3-2 from our releases page.
> 
> 
> 
> Is it available in homebrew yet? I guess I can overlay this version on top but I don’t know what kind of mess that will result with homebrew.

Homebrew/homebrew-cask#79282 has been merged so you can update to 2.5.3-2 using Homebrew.

**Comment by cbednarski** [comment]
2.5.3-2 is working for me on 10.15.4. Thanks for the quick resolution! 🎉 


![Screen Shot 2020-03-25 at 22 31 43](https://user-images.githubusercontent.com/649798/77613647-743f6080-6ee8-11ea-90e0-a044b5615ea8.png)

**Comment by yanns** [comment]
For people using brew:
```
brew cask reinstall keepassxc
```

**Comment by PF93mc8y7erq92qTmTjJBysALa** [comment]
I will tune in with a "same here" though providing the message appearing when launching the app via the terminal:

> open /Applications/KeePassXC.app
> LSOpenURLsWithRole() failed with error -10810 for the file /Applications/KeePassXC.app.

Although KeepassXC was not visible, keepassxc-proxy had been launched and was running.


KeePassXC 2.5.3 installed via homebrew
MacOS 10.15.4

App had previously worked flawlessly under MacOS 10.15.3.


As pointed out above, updating brew then reinstalling the keepassxc cask resulted in keepassxc 2.5.3 working again (having installed _KeePassXC-2.5.3-2.dmg_).


after the reinstall; the version working on macOS 10.15.4
> KeePassXC - Version 2.5.3
> Revision: f8c962b
> 
> Qt 5.14.1
> Diagnosemodus ist deaktiviert.
> 
> Betriebssystem: macOS 10.15
> CPU-Architektur: x86_64
> Kernel: darwin 19.4.0
> 
> Aktivierte Erweiterungen:
> - Auto-Type
> - Browser-Integration
> - SSH-Agent
> - KeeShare (bestätigtes und unbestätigtes Teilen)
> - YubiKey
> - TouchID
> 
> Kryptographische Bibliotheken:
>  libgcrypt 1.8.5
>

**Comment by MaurizioLoreti** [comment]
As a side note for KeePassXC maintainers, one or two days before the installation of Catalina 10.15.4 (i.e. one or two days before KPXC stopped working for me) I tried to select, from the menu, "KeePassXC → Check for Updates". The answer was “2.5.3 is currently the new version available”. In other words, 2.5.3-1 or 2.5.3-2 were not detected by the program itself; and this is a bad behaviour.

--
Maurizio Loreti  --  Maurizio.Loreti@gmail.com

**Comment by mwebber** [comment]
KeePassXC was working for me yesterday
Today it crashes every time when I try to start it

This difference? I have just applied the latest security update to macOS, which means I am now on macOS Hight Sierra 10.13.6 (presumably, yesterday was 10.13.5). Safari was also updated to 13.1.

I suspect one of those updates is what broke KeePassXC

**Comment by phoerious** [comment]
Please read the posts above. There is a download that fixes your issue.

**Comment by mwebber** [comment]
> Please read the posts above. There is a download that fixes your issue.
Thanks, I missed that. 2.5.3-2 works for me.

**Comment by kitzler-walli** [comment]
The new version runs but TouchID doesn't work anymore - should I create a new issue for this?

**Comment by Shev84** [comment]
> The new version runs but TouchID doesn't work anymore - should I create a new issue for this?

Same for me. TouchID doesn't work anymore.

**Comment by grahampugh** [comment]
@kitzler-walli @Shev84 It's a known issue that TouchID does not work when the app is signed and notarised (see https://github.com/keepassxreboot/keepassxc/issues/4398#issuecomment-603705864 above). We will have to wait for the developers to figure out how other apps do it. I would recommend opening a new issue for that. 

This issue (crashing app) is now resolved, although the lack of a version bump means people will most likely keep arriving here looking for solutions until 2.5.4 or 2.6.0 are released.

**Comment by phoerious** [comment]
Our problem is that we cannot enable sandboxing, because then our browser extension would stop working, which is far worse than TouchID not working. I don't see any other solution at the moment and it is probably up to Apple to fix their OS.

**Comment by kitzler-walli** [comment]
I don't need browser integration at all - so TouchID is a bigger showstopper for me. Is there a way to compile the app using sandbox?

**Comment by nmajin** [comment]
> For people using brew:
> 
> ```
> brew cask reinstall keepassxc
> ```

Hmm I did this but it’s still crashing and is still on version 2.5.3; crashes on invalid code signature.

**Comment by PF93mc8y7erq92qTmTjJBysALa** [comment]
@nmajin you might need brew update
in the terminal you should see 
https://github.com/keepassxreboot/keepassxc/releases/download/2.5.3/KeePassXC-2.5.3-2.dmg
being downloaded (notice 2.5.3 **-2** .dmg)

**Comment by nmajin** [comment]
> @nmajin you might need brew update
> 
> in the terminal you should see 
> 
> https://github.com/keepassxreboot/keepassxc/releases/download/2.5.3/KeePassXC-2.5.3-2.dmg
> 
> being downloaded (notice 2.5.3 **-2** .dmg)

Darn, should have know. Thanks, all set, works now.

**Comment by IvanTurgenev** [comment]
> For people using brew:
> 
> ```
> brew cask reinstall keepassxc
> ```

first do a brew uninstall and then brew cleanup

**Comment by eenblam** [comment]
Thanks for the quick response in sorting this out, y'all! Is there a reason this patch can't be released as 2.5.4? Some packaging systems (e.g. AutoPkg + Munki) won't necessarily be able to handle 2.5.3-2 in a predictable way since they assume semantic versioning. Admins can manually import as 2.5.4, but this will later cause a conflict if you later push a release with the same version.

**Comment by grahampugh** [comment]
@eenblam the version is not `2.5.3-2` anyway; that is just the name in the `.dmg`. The version (i.e. `CFBundleShortVersionString`) remains as `2.5.3`, so AutoPkg, Munki, Jamf etc cannot tell the difference.

**Comment by phoerious** [comment]
This is not a patch release. It is the same binary with updated entitlements. A new version would push updates to all platforms despite no code changes whatsoever. Packaging systems are totally able to handle the updated binary. In fact, adding additional numbers for rebuilds of the same upstream version is absolutely common among all major package management systems.

**Comment by grahampugh** [comment]
There are many reasons for bumping a version, which may not include the code in the particular repo from which the release is generated. The release is different, and therefore needs a version bump in order for management systems to recognise the change. You are incorrect that `Packaging systems are totally able to handle the updated binary.` AutoPkg is a very commonly used packaging system and it would not respect the changes you have made without retooling the recipes for this single release. A well-designed packaging tool will take the version of the application (`CFBundleShortVersionString`) and generate a package version from that. In the case of the current version of KeePassXC, the version string in the DMG no longer matches the version of the application, which is bad practice. 

Anyway, I am more speaking pragmatically. There has been almost 40 replies to this issue since you pushed the new version, and I suspect there would be far fewer if you just bumped the version.

**Comment by matthiasschroder** [comment]
Hi guys, I guess you might get less questions and comments about the contents and versioning this release if you put a short message on the download page explaining why this release was necessary, and why you did not bump the version number. My personal assumption is that 2.5.4 was a already assigned to another code change?

**Comment by rpbarlow** [comment]
Many Linux packaging systems have release/build fields in addition to the version field (such as Gentoo or Fedora), where the release is used to indicate that the packaging of the software has changed, but not the source code. Since the MacOS build here was simply rebuilt from the same source code, it makes sense to me that the release/build field would be updated, but not the version.

**Comment by eenblam** [comment]
@phoerius Thanks for the clarification. I can appreciate that it's isolated to Mac releases, and you have other platforms to account for. We'll be able to remediate this on our end without too much, but it's work that will be duplicated by other Mac admin teams. Thanks again for getting a new release signed and notarized quickly.

**Comment by merikan** [comment]
Thanks so much folks, it works fine to reinstall as mentioned above
`brew cask reinstall keepassxc`
Got this problem yesterday when I installed Security Update 2020-002 on MacOS 10.14.6 
Another workaround was to sign the package myself, but I rather run the  official image.

You are awesome! 👍

**Comment by jthomae1** [comment]
The new Version (https://github.com/keepassxreboot/keepassxc/releases/download/2.5.3/KeePassXC-2.5.3-2.dmg) works on macOS 10.15.4

Great! 🎉

**Comment by b-a0** [comment]
The 2.5.3-2 version works, but only after I followed below steps. Perhaps it is also useful to others:

* Open the downloaded `.dmg` file
* Drag the KeePassXC app to the *Desktop*. When dragging to Applications I was greeted with the message that KeePassXC was damaged and should be moved to the bin. Pressing <kbd>Cancel</kbd> just showed the message again.
* After KeePassXS 2.5.3-2 is on the Desktop, open terminal and remove the `com.apple.quarantine` attribute from the `.app` file (steps from [here](https://derflounder.wordpress.com/2012/11/20/clearing-the-quarantine-extended-attribute-from-downloaded-applications/)):
  * `cd ~/Desktop`
  * `sudo xattr -r -d com.apple.quarantine KeePassXC.app`
  * Type the admin password when prompted
* After the quarantine attribute is removed, the app can be moved to the Applications folder and opened normally.

**Comment by droidmonkey** [comment]
Locking this topic since the solution is known and we don't need anymore +1's and me too's. Thank you!
